src/main/java/com/ausregistry/jtoolkit2/se/InetAddress.java

```diff
@@ -55,5 +55,9 @@ public String getTextRep() {
     public Element appendToElement(XMLWriter xmlWriter, Element parent) {
         return xmlWriter.appendChild(parent, "addr", textRep, "ip", getVersion());
     }
+
+    public Element appendHostAddrToElement(XMLWriter xmlWriter, Element parent) {
+        return xmlWriter.appendChild(parent, "hostAddr", textRep, "ip", getVersion());
```

Comment:
`Host` is the only place where it needs to output IP address under the name `hostAddr`. According to the XSD, everywhere else it needs to be output as `addr`. Hence it does not seem necessary to define a new method named `appendHostAddrToElement` in `InetAddress`.

Besides, having two methods, both related with appending element to XML writer in this `InetAddress` class will cause complexity for end users - as they'll need to figure out the differences before they can decide which is the right one to use.

Reply:
Bin, Is it fine if we add like this ?
  public Element appendToElement(XMLWriter xmlWriter, Element parent, String elementName) {
        return xmlWriter.appendChild(parent, elementName, textRep, "ip", getVersion());
    }